Photo filtering repost

quincylvania opened the pull request:
This is #7847 by @nickplesha moved to a branch iD collaborators can commit to.

nickplesha committed:
feat: date and username filtering for photo overlay layers

nickplesha committed:
feat: date and username filtering for photo overlay layers

nickplesha committed:
fix: mapillary sequence filtering

nickplesha committed:
fix: mapillary sequence filtering

nickplesha committed:
feat: filter sequences from openstreetcam and streetside

nickplesha committed:
feat: filter sequences from openstreetcam and streetside

nickplesha committed:
Fix unit tests

nickplesha committed:
fix: fix for filter input values being reset on data loading

quincylvania committed:
Merge pull request #7847 from mapillary/photo-filtering

Filtering for photo overlay layers

quincylvania committed:
Merge branch 'develop' into photo-filter-temp

# Conflicts:
#	css/80_app.css
#	modules/services/mapillary.js

quincylvania committed:
Update date field styling

nickplesha committed:
Persist photo layer filters in the URL

nickplesha commented:
Thanks @quincylvania! Sorry, I was not able to add you to the old branch. I had some work that I forgot to add to the old branch (persisting the active filter values in the URL) before so I've created a new PR for that: https://github.com/openstreetmap/iD/pull/8077

nickplesha committed:
Fix merge conflicts

quincylvania committed:
Merge branch 'develop' into photo-filter-temp

quincylvania committed:
Merge pull request #8077 from mapillary/photo-filtering-url

Persist photo layer filters in the URL

quincylvania committed:
Fix change event for d3 6

quincylvania committed:
Don't show username filter for streetside

quincylvania committed:
Read photo filters from hash before reading enabled overlays/photos

quincylvania committed:
Validate filter date values from hash/input
Compensate if from date is set after to date

quincylvania committed:
Use HTML translated string endpoints for photo option labels

quincylvania committed:
Use localized placeholder for expected date filter format

quincylvania committed:
Reload date filter values after input change

quincylvania committed:
Merge branch 'develop' into photo-filter-temp

quincylvania committed:
Merge branch 'develop' into photo-filter-temp

quincylvania committed:
Combine `fromDate` and `toDate` parameters into single `photo_dates` parameter
Rename `username` parameter to `photo_username`
Add API documentation of photo filter parameters

quincylvania committed:
Disable auto features on photo filter inputs

quincylvania committed:
Don't allow date separators that could be used in some date formats

quincylvania committed:
Support filtering by multiple photo usernames

quincylvania committed:
Make sure photo filter controls are always shown in the same order

quincylvania committed:
Adjust photo filter input styling

quincylvania commented:
Thanks again, @nickplesha! I merged this with a few changes:

- Added fallback functionality for browsers that don't support `date` inputs (like Safari)
- Combined `fromDate` and `toDate` parameters into a single `photo_dates` parameter
- Changed `username` parameter to `photo_username`
- Enabled filtering by multiple comma-separated usernames in both the URL and UI
